src/internet_identity/src/stats/activity_stats.rs

```diff
@@ -82,27 +83,29 @@ pub struct OngoingActivityStats<T: ActivityCounter> {
     pub monthly_events: Vec<T>,
 }
 
-pub fn update_activity_stats(anchor: &Anchor, current_device: &Device) {
+pub fn update_activity_stats(anchor: &Anchor, current_authorization_key: &AuthorizationKey) {
     state::persistent_state_mut(|persistent_state| {
         // Active anchor stats across all domains
         persistent_state
             .active_anchor_stats
             .update_counters(&anchor.last_activity());
 
-        // Active anchor stats, II domains only
-        if let Some(domain) = current_device.ii_domain() {
-            let context = DomainActivityContext {
-                anchor,
-                current_domain: domain,
-            };
-            persistent_state
-                .domain_active_anchor_stats
-                .update_counters(&context);
-
-            // Active authn methods stats, II domains only
-            persistent_state
-                .active_authn_method_stats
-                .update_counters(&current_device);
+        // Anchor device activity across II domains
+        if let AuthorizationKey::DeviceKey(device_key) = current_authorization_key {
+            if let Some(domain) = anchor.device(device_key).and_then(Device::ii_domain) {
+                let context = DomainActivityContext {
+                    anchor,
+                    current_domain: domain,
+                };
+                persistent_state
+                    .domain_active_anchor_stats
```

Comment:
Don't we want to update this for open id?

Does it make sense to have an origin also for open id?

Otherwise, the total active will not match the sum of the domains, right? Or is this tracked somewhere else?

Reply:
I thought that this also was used for the active anchors, but after reviewing it I see this is done in before this in `active_anchor_stats`